CLAMSSpeciesFix.py

```diff
@@ -49,12 +51,14 @@ def __init__(self, parent=None):
                 self.activePartition+"' AND baskets.basket_type='Measure' AND samples.species_code<>0 " +
                 "GROUP BY species.common_name, samples.species_code, species.scientific_name, " +
                 "samples.sample_id,  samples.subcategory",  self.db)
+        query = self.db.dbQuery(sql)
+
         self.sampleDict={}
         self.oldSampleDict={}
         self.speciesCodes=[]
         self.selectionList = []
         while query.next():
```

Comment:
This while block needs to be updated since `query` is now a `dbConnecttion.dbQueryResults` object and NOT a qSQLQuery object. `dbConnecttion.dbQueryResults` is an iterable object and the pattern we're using is to use a for loop to unpack the query results row by row. All values will be returned as strings so the deprecated "to" methods like "toString()" aren't needed. See CLAMSmain.py lines 183-188 for a simple example. Also see [CLAMSProcess.py](https://github.com/noaa-afsc-mace/CLAMS/blob/Py3Qt6/CLAMSprocess.py) for a more thorough example of how the query code blocks are changed.